project.clj

```diff
@@ -229,7 +229,6 @@
             "irb" ["trampoline" "run" "-m" "puppetlabs.puppetserver.cli.irb" "--config" "./dev/puppetserver.conf" "--"]}
 
   :jvm-opts ["-Djruby.logger.class=com.puppetlabs.jruby_utils.jruby.Slf4jLogger"
-             "-XX:+UseG1GC"
```

Comment:
The heap size defaults to something small but prints a warning that you should raise it via a lein user profile. The `heap-size` function called in the next two lines tries to read that before using the smaller defaults.

Reply:
We set the heap explicitly for travis here: https://github.com/puppetlabs/puppetserver/blob/master/ext/travisci/test.sh#L5 (I believe Haus lowered it trying to get fewer OOMs in travis)

Reply:
I would also expect that 9k should now perform similarly to 1.7 from our JRuby work. I think it could be meaningfully slower in tests because of JITing things and constantly tearing down JRuby instances, but I also wouldn't expect the GC to significantly affect that. I'll test it locally and report back.

Reply:
I'll be damned. G1GC has a significant affect on our 9k test times.


Normal dev settings with this PR on 1.7
```
sellout:puppetserver justin$ time lein test :all
Using 1G min heap since not set via PUPPETSERVER_HEAP_SIZE environment variable or user.puppetserver-heap-size in ~/.lein/profiles.clj file. Set to at least 5G for best performance during test runs.
...snip...
Ran 190 tests containing 2480 assertions.
1 failures, 0 errors.
Tests failed.

real    25m36.441s
user    133m42.308s
sys     1m22.903s
```
With 6gs of memory on 1.7
```
sellout:puppetserver justin$ PUPPETSERVER_HEAP_SIZE=6G time lein test :all
Compiling ClojureScript...
Compiling ClojureScript...
...snip...
Ran 190 tests containing 2480 assertions.
1 failures, 0 errors.
Tests failed.
      582.84 real      1361.41 user        67.84 sys
```
Using 6gs of memory with the last released JRuby9k deps
```
sellout:puppetserver justin$ PUPPETSERVER_HEAP_SIZE=6G time lein with-profile +jruby9k test :all
Compiling ClojureScript...
Compiling ClojureScript...
...snip...
Ran 190 tests containing 2480 assertions.
1 failures, 0 errors.
Tests failed.
Error encountered performing task 'test' with profile(s): 'base,system,user,provided,dev,jruby9k'
Tests failed.
     2656.66 real      8208.86 user       504.85 sys
```
And running the same 9k with the G1GC:
```
sellout:puppetserver justin$ git revert f3cd9c8
[bump9k 3c76fc6] Revert "(maint) Do not use G1GC in development"
 1 file changed, 1 insertion(+)
sellout:puppetserver justin$ PUPPETSERVER_HEAP_SIZE=6G time lein with-profile +jruby9k test :all
Compiling ClojureScript...
...snip...
Ran 190 tests containing 2480 assertions.
1 failures, 0 errors.
Tests failed.
Error encountered performing task 'test' with profile(s): 'base,system,user,provided,dev,jruby9k'
Tests failed.
     1763.71 real      3228.16 user       432.97 sys
```

Reply:
:/ not great, but yeah, improvement worth having